### DISCUSSION TITLE
How coud i get baggage in SpanExporter?

### DISCUSSION THREAD

**Discussion by mijingling** [open-telemetry/opentelemetry-java-instrumentation#3877] [Q&A]
**Is your feature request related to a problem? Please describe.**

I could get baggage in application, but i can't get baggage in SpanExporter
How coud i get baggage in SpanExporter?

## get baggage in SpanExporter
public class MySpanExporter implements SpanExporter {
	@Override
	public CompletableResultCode export(Collection<SpanData> spans) {
		_String tradeCode = Baggage.current().getEntryValue("trade_code");_
		**// but tradeCode is null , How coud i get baggage in SpanExporter?**
		return CompletableResultCode.ofSuccess();
	}
	...
}

thanks very much^^

**Comment by jkwatson** [comment]
Hi @mijingling . There is no way to get access to Baggage in an exporter. If you need to append Baggage entries to a Span, the best place for that is in a SpanProcessor, which does have access to the span's Context, and hence the Baggage.
